nexus/src/app/update/mod.rs

```diff
@@ -56,17 +56,30 @@ impl super::Nexus {
             ArtifactsWithPlan::from_stream(body, Some(file_name), &self.log)
                 .await
                 .map_err(|error| error.to_http_error())?;
-
-        // Now store the artifacts in the database.
         let tuf_repo_description = TufRepoDescription::from_external(
             artifacts_with_plan.description().clone(),
         );
-
+        // Move the `ArtifactsWithPlan`, which carries with it the
+        // `Utf8TempDir`s storing the artifacts, into the artifact replication
+        // background task. This is done before the database insert because if
+        // this fails, we shouldn't record the artifacts in the database.
+        self.tuf_artifact_replication_tx
```

Comment:
We're sending the artifacts on this channel *before* we write them into the DB -- if the background task activated after we write to this mpsc, but before we call `update_tuf_repo_insert`, would discard the TUF artifacts we just tried to add?

Reply:
Hmm. I think it would. This seems clearly worse than "we recorded the artifacts in the database, but failed to move the artifacts into the mpsc", right? Particularly since failing to move the artifacts shouldn't ever really happen unless somehow the background tasks are dead.

Probably I will flip these around, which also gives me the opportunity to avoid sending the artifacts and activating the task if the repo is either already uploaded or was rejected for whatever reason.